### PR TITLE
BHBC-1077: Add support for spatial overlay layers

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-1077

## This PR contains the following changes

Add 3 spatial overlays to the map.
Add support for overlays in general.
Added popup item to overlay features (mainly to demonstrate the concept) with button to add that feature to the geometry state.

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: app/src/components/map/MapContainer.tsx [hunk 1]
```diff
@@ -87,6 +103,32 @@ const MapContainer: React.FC<IMapContainerProps> = (props) => {
         <LayersControl.BaseLayer name="BC Government">
           <TileLayer url="https://maps.gov.bc.ca/arcgis/rest/services/province/roads_wm/MapServer/tile/{z}/{y}/{x}" />
         </LayersControl.BaseLayer>
+        {hideOverlayLayers !== true && (
```
Comment (sdevalapurkar) [inline app/src/components/map/MapContainer.tsx]: nit
can we just say: 
```suggestion
        {!hideOverlayLayers && (
```

File: app/src/components/map/MapContainer.tsx [hunk 2]
```diff
@@ -87,6 +103,32 @@ const MapContainer: React.FC<IMapContainerProps> = (props) => {
...
+          <>
+            <LayersControl.Overlay name="Wildlife Management Units">
+              <WFSFeatureGroup
+                name={'Wildlife Management Units'}
+                typeName={'pub:WHSE_WILDLIFE_MANAGEMENT.WAA_WILDLIFE_MGMT_UNITS_SVW'}
+                minZoom={8}
```
Comment (sdevalapurkar) [inline app/src/components/map/MapContainer.tsx:122]: question: why is minzoom here different than other predefined geometries?

File: app/src/components/map/WFSFeatureGroup.test.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  external: {
+    get: jest.fn()
+  }
+};
+
+const mockBiohubApi = ((useBiohubApi as unknown) as jest.Mock<typeof mockUseBiohubApi>).mockReturnValue(
+  mockUseBiohubApi
+);
+
+describe('MapContainer', () => {
```
Comment (sdevalapurkar) [inline app/src/components/map/WFSFeatureGroup.test.tsx]:
```suggestion
describe('WFSFeatureGroup.test', () => {
```

File: app/src/components/map/MapContainer.tsx (hunk 2 above)
Comment (NickPhura) [reply app/src/components/map/MapContainer.tsx:122]: Arbitrarily chosen by me, but its based on how large the features are, and how many there are.

If the features are large, and there are fewer, then a lower zoom level is fine.  But some layers have a huge number of small features, so a higher zoom level would be better.

File: app/src/components/map/MapContainer.tsx (hunk 1 above)
Comment (NickPhura) [reply app/src/components/map/MapContainer.tsx]: fixed

File: app/src/components/map/WFSFeatureGroup.test.tsx (hunk 3 above)
Comment (NickPhura) [reply app/src/components/map/WFSFeatureGroup.test.tsx]: fixed

File: app/src/components/map/MapContainer.test.tsx [hunk 4]
```diff
@@ -81,6 +102,62 @@ describe('MapContainer', () => {
     expect(asFragment()).toMatchSnapshot();
   });
 
+  test.only('updates the geometry when a pre-defined feature is returned by an overlay layer', async () => {
```
Comment (sdevalapurkar) [inline app/src/components/map/MapContainer.test.tsx]: remove only
```suggestion
  test('updates the geometry when a pre-defined feature is returned by an overlay layer', async () => {
```

**Review by anissa-agahchen — DISMISSED** [review]

File: app/src/components/map/MapContainer.test.tsx (hunk 4 above)
Comment (NickPhura) [reply app/src/components/map/MapContainer.test.tsx]: nice catch, updated

**Review by sdevalapurkar — APPROVED** [review]
nicely done
download

**Pull request merged**